staging/src/k8s.io/kubectl/pkg/cmd/get/get.go

```diff
@@ -813,7 +813,6 @@ func (o *GetOptions) printGeneric(r *resource.Result) error {
 		}
 		if listMeta, err := meta.ListAccessor(obj); err == nil {
 			list.Object["metadata"] = map[string]interface{}{
-				"selfLink":        listMeta.GetSelfLink(),
```

Comment:
I would still make this conditional so that if we're talking to an old server we don't drop the field before the server does :)

```go
if selfLink := listMeta.GetSelfLink(); len(selfLink) > 0 {
  list.Object["metadata"]["selfLink"] = selfLink
}
```

Reply:
I understand your concern, but I'm not sure I fully agree with you.
Basically, the reason why we decided to deprecate selflink is not just it was causing us some issues, but most of all that noone was really using it for anything useful. So given that their apiserver will sooner or later stop setting selflinks, do we really still need to print it?
If someone really needs it, they are still able to get it via -o json or -o yaml or etc.

Reply:
hmm... since the field stopped being populated by default well outside the stated kubectl skew, I guess simply dropping is ok.

if the server had just stopped populating this release, I'd still push to make this conditional

Reply:
OK yeah - I should have mentioned that too, I agree this point is important.